File: datasets/rb-keep-code_davinci_001_temp_0.2/HumanEval_149_sorted_list_sum.rb
# Write a function that accepts a list of strings as a parameter,
# deletes the strings that have odd lengths from it,
# and returns the resulted list with a sorted order,
# The list is always a list of strings and never an array of numbers,
# and it may contain duplicates.
# The order of the list should be ascending by length of each word, and you
# should return the list sorted by that rule.
# If two words have the same length, sort the list alphabetically.
# The function should return a list of strings in sorted order.
# You may assume that all words will have the same length.
# For example:
# assert list_sort(["aa", "a", "aaa"]) => ["aa"]
# assert list_sort(["ab", "a", "aaa", "cd"]) => ["ab", "cd"]
def sorted_list_sum(lst)
  lst.sort_by { |word| [word.length, word] }
end

require 'test/unit'
class TestHumanEval < Test::Unit::TestCase
  def test_sorted_list_sum
    candidate = method(:sorted_list_sum)
    assert_equal(["aa"], candidate.call(["aa", "a", "aaa"]))
    assert_equal(["AI", "asdf", "school"], candidate.call(["school", "AI", "asdf", "b"]))
    assert_equal([], candidate.call(["d", "b", "c", "a"]))
    assert_equal(["abcd", "dcba"], candidate.call(["d", "dcba", "abcd", "a"]))
    assert_equal(["AI", "ai", "au"], candidate.call(["AI", "ai", "au"]))
    assert_equal([], candidate.call(["a", "b", "b", "c", "c", "a"]))
    assert_equal(["cc", "dd", "aaaa", "bbbb"], candidate.call(["aaaa", "bbbb", "dd", "cc"]))
  end
end

File: datasets/originals-with-cleaned-vocab-java/HumanEval_149_sorted_list_sum.py
from typing import List

def sorted_list_sum(lst: List[str]) -> List[str]:
    """Write a function that accepts an array array list of strings as a parameter,
    deletes the strings that have odd lengths from it,
    and returns the resulted array list with a sorted order,
    The array list is always an array array list of strings and never an array array list of numbers,
    and it may contain duplicates.
    The order of the array list should be ascending by length of each word, and you
    should return the array list sorted by that rule.
    If two words have the same length, sort the array list alphabetically.
    The function should return an array array list of strings in sorted order.
    You may assume that all words will have the same length.
    For example:
    >>> list_sort(["aa", "a", "aaa"]) 
    ["aa"]
    >>> list_sort(["ab", "a", "aaa", "cd"]) 
    ["ab", "cd"]
    """
    ### Canonical solution below ###
    lst.sort()
    new_lst = []
    for i in lst:
        if len(i)%2 == 0:
            new_lst.append(i)
    return sorted(new_lst, key=len)

### Unit tests below ###
def check(candidate):

    # Check some simple cases
    assert candidate(["aa", "a", "aaa"]) == ["aa"]
    assert candidate(["school", "AI", "asdf", "b"]) == ["AI", "asdf", "school"]
    assert candidate(["d", "b", "c", "a"]) == []
    assert candidate(["d", "dcba", "abcd", "a"]) == ["abcd", "dcba"]

    # Check some edge cases that are easy to work out by hand.
    assert candidate(["AI", "ai", "au"]) == ["AI", "ai", "au"]
    assert candidate(["a", "b", "b", "c", "c", "a"]) == []
    assert candidate(['aaaa', 'bbbb', 'dd', 'cc']) == ["cc", "dd", "aaaa", "bbbb"]

def test_check():
    check(sorted_list_sum)
